index.js

```diff
@@ -95,40 +102,148 @@ class instance extends instance_skel {
+				for (let index = 0; index < this.outputCount; index++) {
+					data[index] = index  + ' ' + this.getOutput(index).route;
+					string = string + index + "  " + this.getOutput(index).fallback + "\n"
+				}
+				try{
+					// use the blocking version of file write here.
+					fs.writeFileSync(opt.destination_file, data + string, 'utf8');
+					this.log('info',data.length + " Routes written to: " + opt.destination_file );
+				}
+				catch (e) {
+					this.log('error',"File Write Error: " + e.message);
+				}
+				break;
+
+			case 'load_route_from_file': // added ability to read routes from a file v1.3.0
+
+				try {
+					// use the blocking version of file read here. 
+					var data = fs.readFileSync(opt.source_file, 'utf8');
+					try{
+						routes_text= data.split(':');  // strip out unwanted text after the ':'
+						routes = routes_text[0].split(',');  // split routes into source - destination array
+						if((routes.length > 0) && (routes.length <= this.outputCount)) {      
+						    cmd = '';
+							for (let index = 0; index < routes.length; index++) {
+								dest_source = routes[index].split(' ');
+								if(isNaN(dest_source[0])) {
+									throw routes[index] + " - " + dest_source[0] + " is not a valid Router Destination ";
+								}
+								if(dest_source[0] < 0 || dest_source[0] > (this.outputCount - 1)) {
+									throw dest_source[0] + "  is an invalid destination.  Remember, Router is zero based when indexing ports.  Max Routes for this router = " + this.outputCount;
+								}
+								if(isNaN(dest_source[1])) {
+									throw routes[index] + " - " + dest_source[1] + " is not a valid Router Source ";
+								}
+								if(dest_source[1] < 0 || dest_source[1] > (this.outputCount - 1)) {
+									throw dest_source[1] + "  is an invalid source. Remember, Router is zero based when indexing ports.  Max Routes for this router = " +this.outputCount; 
+								}
+								cmd = cmd + "VIDEO OUTPUT ROUTING:\n" +  routes[index] + "\n\n";  // not sure if we need both \n here...
+							}
+						}
+						else {
+							throw "Invalid number of Routes: " + routes.length + ",";
+						}
+						this.log('info', routes.length + " Routes read from File: " + opt.source_file );
+					}
+					catch (err) {
+						this.log('error', err + " in File:" + opt.source_file);
+					}
+				}
+				catch(e) {
+					this.log('error',"File Read Error: " + e.message);
+				}
+				break;
+
 			case 'route':
+
+				var output = this.getOutput(parseInt(opt.destination));
+
 				if (parseInt(opt.destination) >= this.outputCount) {
 					cmd = "VIDEO MONITORING OUTPUT ROUTING:\n"+(parseInt(opt.destination)-this.outputCount)+" "+opt.source+"\n\n";
 				}
 				else {
 					cmd = "VIDEO OUTPUT ROUTING:\n"+opt.destination+" "+opt.source+"\n\n";
 				}
 				break;
+
+			case 'route_to_previous':
+
+				var output = this.getOutput(parseInt(opt.destination));
+				var fallbackpop = -1;
+
+				if(output !== undefined){
```

Comment:
Although this check is fine and good practice, a careful reading of `getOutput` would show it always provides a return.